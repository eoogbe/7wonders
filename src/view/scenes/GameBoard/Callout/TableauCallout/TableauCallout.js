/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { getTableauCard } from 'presentation/tableaus';
import CalloutSection from '../CalloutSection';
import '../Callout.css';

const TableauCallout = ({ cardId }) => {
  const { name, type, effects, chainedStructures } = getTableauCard(cardId);

  return (
    <section>
      <h2 className="Callout-title">{name}</h2>
      <p className="Callout-paragraph">Type: {type}</p>
      <CalloutSection heading="This card awards" items={effects} />
      {chainedStructures.length > 0 &&
        <CalloutSection
          heading="Having this card allows the player to build for free"
          items={chainedStructures}
        />}
    </section>
  );
};

export default TableauCallout;
