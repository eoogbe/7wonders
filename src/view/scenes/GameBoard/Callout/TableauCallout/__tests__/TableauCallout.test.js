/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import TableauCallout from '../TableauCallout';

it('renders without crashing', () => {
  const tree =
    renderer.create(<TableauCallout cardId={'LUMBER_YARD'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with chained structures', () => {
  const tree =
    renderer.create(<TableauCallout cardId={'APOTHECARY'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
