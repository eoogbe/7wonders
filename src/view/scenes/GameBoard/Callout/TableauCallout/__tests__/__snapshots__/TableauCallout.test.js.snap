// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with chained structures 1`] = `
<section>
  <h2
    className="Callout-title"
  >
    Apothecary
  </h2>
  <p
    className="Callout-paragraph"
  >
    Type: 
    Science
  </p>
  <section
    className="CalloutSection"
  >
    <h3
      className="CalloutSection-heading"
    >
      This card awards
      :
    </h3>
    <ul
      className="CalloutSection-list"
    >
      <li>
        Compass
      </li>
    </ul>
  </section>
  <section
    className="CalloutSection"
  >
    <h3
      className="CalloutSection-heading"
    >
      Having this card allows the player to build for free
      :
    </h3>
    <ul
      className="CalloutSection-list"
    >
      <li>
        Dispensary
      </li>
      <li>
        Stables
      </li>
    </ul>
  </section>
</section>
`;

exports[`renders without crashing 1`] = `
<section>
  <h2
    className="Callout-title"
  >
    Lumber yard
  </h2>
  <p
    className="Callout-paragraph"
  >
    Type: 
    Raw material
  </p>
  <section
    className="CalloutSection"
  >
    <h3
      className="CalloutSection-heading"
    >
      This card awards
      :
    </h3>
    <ul
      className="CalloutSection-list"
    >
      <li>
        1 Wood
      </li>
    </ul>
  </section>
</section>
`;
